services/placement/types.go

```diff
@@ -36,13 +36,39 @@ type Algorithm interface {
 	// RemoveInstances removes a list of instances from the placement.
 	RemoveInstances(p services.Placement, leavingInstanceIDs []string) (services.Placement, error)
 
-	// ReplaceInstance replace a instance with new instances.
-	ReplaceInstance(p services.Placement, leavingInstanceID string, addingInstances []services.PlacementInstance) (services.Placement, error)
+	// ReplaceInstance replace a list of instances with new instances.
+	ReplaceInstances(
```

Comment:
nit:  retain the old method, and have it call the new extended method. will be good to not break all the dependent code (e.g. m3admin, dtests)